### ISSUE TITLE
DD (DoubleDouble) parsing issue with whole numbers

### ISSUE THREAD

**Issue by FObermaier** [locationtech/jts#260]
Parsing negative whole numbers with DD.Parse yields wrong results.

```Java
assertEquals(DD.valueOf(1), DD.parse("1")); // DD.parse yields 0.1
assertEquals(DD.valueOf(100), DD.parse("100")); // DD.parse yields 0.1
assertEquals(DD.valueOf(-1), DD.parse("-1")); // DD.parse yields -0.1
assertEquals(DD.valueOf(-100), DD.parse("-100")); // DD.parse yields -0.1
assertEquals(DD.valueOf(-1), DD.parse("-123")); // DD.parse yields -0.123
```

Please see attached patchfile for extended DDIOTest unit test.
[Added_test_for_DD_parse_of_whole_numbers.zip](https://github.com/locationtech/jts/files/1743622/Added_test_for_DD_parse_of_whole_numbers.zip)

**Comment by FObermaier** [comment]
The would be the fix

```Java
    // somewhere around line 1358 
    DD val2 = val;

    // correct number of digits before decimal sign if we have a whole negative number
    if (numBeforeDec == 0 && isNegative) numBeforeDec = 1;

    // scale the number correctly
    int numDecPlaces = numDigits - numBeforeDec - exp;

```

**Comment by FObermaier** [comment]
This would be the fix:

```Java
    DD val2 = val;

    // correct number of digits before decimal sign if we have a whole negative number
    if (numBeforeDec == 0 && isNegative) numBeforeDec = numDigits;

    // scale the number correctly
    int numDecPlaces = numDigits - numBeforeDec - exp;
```

**Comment by dr-jts** [comment]
Fixed by #260
